workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/LayoutScreenViewFactory.kt

```diff
@@ -0,0 +1,36 @@
+package com.squareup.workflow1.ui
+
+import android.content.Context
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import kotlin.reflect.KClass
+
+/**
+ * A [ScreenViewFactory] that ties a [layout resource][layoutId] to a
+ * [ViewRunner factory][runnerConstructor] function. See [ScreenViewRunner] for
+ * details.
+ */
+@WorkflowUiExperimentalApi
+@PublishedApi
+internal class LayoutScreenViewFactory<RenderingT : Screen>(
```

Comment:
Tempted to change all of these `RenderingT` param types to `ScreenT`. Downside is that we lose the hint that these are expected to be the `RenderingT` from a workflow. Probably won't mess.